libs/crypto/hash.hpp

```diff
@@ -48,39 +48,39 @@ namespace iroha {
       return SHA3_512(output, input, in_size);
     }
 
-    nonstd::optional<std::vector<uint8_t>> sha3_224(uint8_t *input,
+    inline nonstd::optional<std::vector<uint8_t>> sha3_224(uint8_t *input,
```

Comment:
The functions, `nonstd::optional<> sha3_...()`, seem simply wrapping the features of `SHA3_...()`
Ofcourse, guranteeing correct input is always required. Hmm... in this situation, optional may be redundant.